src/txn/txn.c

```diff
@@ -964,32 +953,30 @@ __txn_fixup_prepared_update(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, WT_U
      */
     F_SET(txn, WT_TXN_PREPARE_IGNORE_API_CHECK);
 
+    /*
+     * Set the stop time point to be the committing transaction's time point and copy the start time
+     * point from the current history store update.
+     */
     tw.stop_ts = txn->commit_timestamp;
     tw.durable_stop_ts = txn->durable_timestamp;
     tw.stop_txn = txn->id;
-    WT_TIME_WINDOW_SET_START(&tw, fix_upd);
+    tw.start_ts = hs_tw->start_ts;
+    tw.durable_start_ts = hs_tw->durable_start_ts;
+    tw.start_txn = hs_tw->start_txn;
 
-#ifdef HAVE_DIAGNOSTIC
```

Comment:
We can still can that check. I removed it because we already checked this before

```
    /*
     * If the history update already has a stop time point there is no work to do. This happens if a
     * deleted key is reinserted by a prepared update.
     */
    if (WT_TIME_WINDOW_HAS_STOP(hs_tw))
        return (0);
```

Reply:
Fixed